### PR TITLE
OS internal function should indicate the error by return negative value

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
instead to change errno value by calling set_errno

## Impact
No function change.

### CODE REVIEW

**Review by Ouss4 — APPROVED** [review]

File: drivers/sensors/adxl372.c [hunk 1]
```diff
@@ -716,10 +720,12 @@ static void adxl372_dvr_exchange(FAR void *instance_handle,
                                  FAR const void *txbuffer,
                                  FAR void *rxbuffer, size_t nwords)
 {
-  FAR struct adxl372_dev_s *priv = (FAR struct adxl372_dev_s *)instance_handle;
+  FAR struct adxl372_dev_s *priv = instance_handle;
```
Comment (Ouss4) [inline drivers/sensors/adxl372.c:723]: Why is the implicit cast removed here?

**Comment by Ouss4** [comment]
BTW the checks are failing because of #2618.  The following should fix the issue but I don't know if there is a better solution.

```diff
diff --git a/fs/vfs/fs_unlink.c b/fs/vfs/fs_unlink.c
index 5b0b6053f7..490fcd0059 100644
--- a/fs/vfs/fs_unlink.c
+++ b/fs/vfs/fs_unlink.c
@@ -191,8 +191,10 @@ int nx_unlink(FAR const char *pathname)
   RELEASE_SEARCH(&desc);
   return OK;

+#ifndef CONFIG_DISABLE_PSEUDOFS_OPERATIONS
 errout_with_sem:
   inode_semgive();
+#endif
```

**Comment by Ouss4** [comment]
We have #2629 for that issue.

File: drivers/sensors/adxl372.c (hunk 1 above)
Comment (xiaoxiang781216) [reply drivers/sensors/adxl372.c:723]: To fix the long length warning.

**Comment by xiaoxiang781216** [comment]
> We have #2629 for that issue.

Thanks, I will rebase to the latest mainline.

File: drivers/sensors/adxl372.c (hunk 1 above)
Comment (xiaoxiang781216) [reply drivers/sensors/adxl372.c]: @Ouss4 I short instance_handle to instance, please review again.

**Review by Ouss4 — APPROVED** [review]

**Pull request merged**